app/controllers/users_controller.rb

```diff
@@ -151,7 +151,17 @@ def crop_image_and_save(user, avatar)
     )
   end
 
-  def content_type_permitted(avatar_content_type)
-    %w[image/jpeg image/pjpeg image/png image/x-png].include? avatar_content_type
+  def validate_avatar(file)
```

Comment:
have you considered moving this to a Validator class? this we wouldn't clutter the controller with a new responsibility and it could be tested in isolation.

Reply:
Hi @sauloperez, yes I considered moving it to validators (via https://github.com/igorkasyanchuk/active_storage_validations), but it needs more "rewrites", as the endpoint that crops and prepares the image is not a regular `user.update(params)`. In fact, we don't save the user object at all. 

Probably a good compromise is to move all this avatar operations (2 methods in this controller) to a service class, ala: `AvatarGenerator`.

Reply:
definitely, the 1st step is to move this out of the controller. Also, we could also come up with an `AvatarValidator` of our own implemented as a simple PORO.

Reply:
> the 1st step is to move this out of the controller

will do it in this branch!

@sauloperez ℹ️  for the sake of reference, there is an amazing gem to unify Services/Commands interfaces in Ruby: https://github.com/markets/serviz ✨ ✨ 😺

Reply:
which incidentally happens to be written by you :joy: it looks nice though :ok_hand:

Reply:
😂  you caught me! But I can promise is the most minimalistic interface you'll need (and only 30LOC) to homogenize all your Services:

- `#call` -> invoke
- `#success?`/ `#failure?` -> ended well or bad?
- `#result` -> get the result
- `#errors` -> get the errors